pyquil/api/tests/test_base_connection.py

```diff
@@ -0,0 +1,336 @@
+import pytest
+import requests_mock
+import urllib.parse
+
+from pyquil.api._base_connection import ForestSession
+from pyquil.api._config import PyquilConfig
+from pyquil.api._errors import UserMessageError
+from pyquil.tests.utils import api_fixture_path
+
+
+@pytest.fixture
+def successful_engagement_response():
+    return {
+        'engage': {
+            'success': True,
+            'message': "Good job, you engaged all by yourself",
+            'engagement': {
+                'type': 'RESERVATION',
+                'qpu': {
+                    'endpoint': 'tcp://fake.url:12345',
+                    'credentials': {
+                        'clientPublic': 'abc123',
+                        'clientSecret': 'abc123',
+                        'serverPublic': 'abc123'
+                    }
+                },
+                'compiler': {
+                    'endpoint': 'tcp://fake.url:12346'
+                },
+                'expiresAt': '9999999999.0'
+            }
+        }
+    }
+
+
+@pytest.fixture
+def failed_engagement_response():
+    return {'engage': {'success': False, 'message': "That did not work", 'engagement': None}}
+
+
+test_config_paths = {
```

Comment:
like this for example -- its a "test fixture" but doesnt need to be declared as a `pytest.fixture`